app/groups/[id].tsx

```diff
@@ -0,0 +1,44 @@
+import { useLocalSearchParams, useNavigation } from 'expo-router';
+import { useEffect } from 'react';
+import { Text, View } from 'react-native';
+import { getGroupById } from '../data/groups';
+
+export default function GroupDetailsScreen() {
+  const params = useLocalSearchParams<{ id?: string }>();
+  const navigation = useNavigation();
+
+  const group = params.id ? getGroupById(String(params.id)) : undefined;
```

Comment:
do we want to make this param optional?

Reply:
Discussed live, we don't need this now, but in case we do come across this issue, we at least have a fallback. We should keep in mind the cases where we run into this though.